scripts/cnode-helper-scripts/gLiveView.sh

```diff
@@ -768,6 +770,21 @@ printf "${NC}"       # reset and set default color
 
 unset cpu_now cpu_last
 
+####################################
+# Mithril Signer Section Variables #
+####################################
+
+# mithril.env sourcing needed to have values in ${METRICS_SERVER_IP} and ${METRICS_SERVER_PORT}
+. ${CNODE_HOME}/mithril/mithril.env
+
+serviceExists=$(systemctl list-units --type=service --all | grep -q ${CNODE_VNAME}-mithril-signer.service; echo $?)
+
+mithrilSignerVars() {
+signerServiceStatus=$(systemctl status ${CNODE_VNAME}-mithril-signer.service 2>/dev/null | grep 'Active:' | awk '{print $3}' | sed 's/[()]//g')
+metricsEnabled=$(grep -q "ENABLE_METRICS_SERVER=true" ${CNODE_HOME}/mithril/mithril.env && echo "true" || echo "false")
+mithrilSignerMetrics=$(curl -s "http://${METRICS_SERVER_IP}:${METRICS_SERVER_PORT}/metrics" 2>/dev/null | grep -v -E "HELP|TYPE" | sed 's/mithril_signer_//g')
+}
```

Comment:
I've tested and it looks good and mithril script changes work as intended. I've been contemplating the way to provide support for non Systemd environments. Two thoughts come to mind. 
1. **MITHRIL_SIGNER_TYPE** `(SYSTEMD|OTHER) Type of Mithril Signer Environment`
   * This keeps the current logic where systemd service unit and service status checks are used.
   * When OTHER (anything besides SYSTEMD) is set, only the metrics port responsiveness is used for service status.
   * Unset by default.
2. **MITHRIL_SIGNER_ENABLED** `(Y|N) If the Mithril Signer is enabled`
   * Drops the specific logic for Systemd, and only relies on the metrics server port and responsiveness.
   * Unset by default, but logic defaults to `MITHRIL_SIGNER_ENABLED=N` when variable is not set.

```
mithrilSignerVars() {
  signerMetricsEnabled=$(grep -q "ENABLE_METRICS_SERVER=true" ${CNODE_HOME}/mithril/mithril.env && echo "true" || echo "false")
  echo "singerMetricsEnabled: ${signerMetricsEnabled}" >> /tmp/mithril-gliveview.log
  if [[ "${signerMetricsEnabled}" == "true" ]] ; then
    mithrilSignerMetrics=$(curl -s "http://${METRICS_SERVER_IP}:${METRICS_SERVER_PORT}/metrics" 2>/dev/null | grep -v -E "HELP|TYPE" | sed 's/mithril_signer_//g')
    echo "mithrilSignerMetrics: ${mithrilSignerMetrics}" >> /tmp/mithril-gliveview.log
    if [[ "${MITHRIL_SIGNER_TYPE}" == "SYSTEMD" ]] ; then 
      echo "MITHRIL_SIGNER_TYPE: ${MITHRIL_SIGNER_TYPE}" >> /tmp/mithril-gliveview.log

      signerServiceStatus=$(systemctl status ${CNODE_VNAME}-mithril-signer.service 2>/dev/null | grep 'Active:' | awk '{print $3}' | sed 's/[()]//g')
      echo "signerServiceStatus: ${signerServiceStatus}" >> /tmp/mithril-gliveview.log
    elif [[ -n "${MITHRIL_SIGNER_TYPE}" ]] && [[ ! "${MITHRIL_SIGNER_TYPE}" == "SYSTEMD" ]] ; then
      SIGNER_METRICS_HTTP_RESPONSE=$(curl --write-out "%{http_code}" --silent --output /dev/null --connect-timeout 2 http://${METRICS_SERVER_IP}:${METRICS_SERVER_PORT}/metrics)
      echo "SIGNER_METRICS_HTTP_RESPONSE: ${SIGNER_METRICS_HTTP_RESPONSE}" >> /tmp/mithril-gliveview.log
      if [[ "$SIGNER_METRICS_HTTP_RESPONSE" -eq 200 ]] ; then
        signerServiceStatus='running'
      else
        signerServiceStatus='inactive (dead)'
      fi
      unset SIGNER_METRICS_HTTP_RESPONSE
    fi
  fi
}
```

This change skips trying to curl the metrics server when the metrics are not enabled via the variable. Later the check would drop `serviceExists` an use instead `[[ -n "${MITHRIL_SIGNER_TYPE}" ]]`.

```
      if [[ "$nodemode" = "Core" ]] && [[ -n "${MITHRIL_SIGNER_TYPE}" ]] && [[ "$signerMetricsEnabled" == "true" ]] && [[ "$signerServiceStatus" == "running" ]] ; then
```

Given Systemd and non systemd (manual, containerized, etc.) all rely on the fact that the metrics server has to be used I'm a bit partial to the second solution. The number of branches required in **mithrilSignerVars** would be reduced quite a bit as when the metrics server is not responding the Mithril Signer status could be set to offline, no need to check the systemd service. However, I only provided code suggestion for the first option to keep it aligned with the current PR logic even though it takes more lines/logic to achieve.

@asnakep What are your thoughts?

@rdlrt @Scitz0 Do you have any preference between the two options mentioned? Does a third option come to mind?

Reply:
This is untested, but I quickly tossed together what I think the function would look like if using **MITHRIL_SIGNER_ENABLED** and not relying on systemd, but simply the metrics server response.

```
mithrilSignerVars() {
  signerMetricsEnabled=$(grep -q "ENABLE_METRICS_SERVER=true" ${CNODE_HOME}/mithril/mithril.env && echo "true" || echo "false")
  if [[ "${MITHRIL_SIGNER_ENABLED}" == "Y" ]] && [[ "${signerMetricsEnabled}" == "true" ]] ; then
    mithrilSignerMetrics=$(curl -s "http://${METRICS_SERVER_IP}:${METRICS_SERVER_PORT}/metrics" 2>/dev/null | grep -v -E "HELP|TYPE" | sed 's/mithril_signer_//g')
    SIGNER_METRICS_HTTP_RESPONSE=$(curl --write-out "%{http_code}" --silent --output /dev/null --connect-timeout 2 http://${METRICS_SERVER_IP}:${METRICS_SERVER_PORT}/metrics)
    if [[ "$SIGNER_METRICS_HTTP_RESPONSE" -eq 200 ]] ; then
      signerServiceStatus='running'
    else
      signerServiceStatus='inactive (dead)'
    fi
    unset SIGNER_METRICS_HTTP_RESPONSE
  fi
}
```

Reply:
Rather than extensive checks every time, regardless of whether mithril signer is being run - I'd prefer to have a variable at the top about enabling Mithril section (defaulting to 'N' ) and then only when the value for that option is not 'Y' , to start looking at options. Thus, your suggestion wrt `MITHRIL_SIGNER_ENABLED` makes sense - just need to ensure this part.

As regards with systemd v/s without, I dont want to put too many combinations in, if enabled and running on a particular port, best to simply use IP:port as a test, exclude systemd completely from code where possible - as it wont be present on docker systems anyways

Reply:
Hi Trevor and CNTools/Koios Team,

I strongly agree with your additions on check metrics response, mithril signer enabled var and signer type, just please let me know if some action is required from my side, or if this changes will be applied by you.

Thank you
Cheers,
Manuel

Reply:
I'll make some adjustments and retest for MITHRIL_ENABLED. Submit a new commit later today after some additional testing.

Reply:
Change now gates all steps based on **MITHRIL_SIGNER_ENABLED** being Y, as per @rdlrt suggestion we won't parse files or collect variables unless this is set, even when the metrics server is enabled by the SPO.

This also allows the SPO to run a mithril signer, but decide they want the original gLiveView layout by changing this value to N, without impacting the signer itself or their ability to query its port separately. Would require an update somewhere in documentation to describe it.

```suggestion
mithrilSignerVars() {
  # Require MITHRIL_SIGNER_ENABLED or do not proceed to source files or check ports.
  if [[ "${MITHRIL_SIGNER_ENABLED}" == "Y" ]] ; then
    # mithril.env sourcing needed to have values in ${METRICS_SERVER_IP} and ${METRICS_SERVER_PORT}
    . ${CNODE_HOME}/mithril/mithril.env
    signerMetricsEnabled=$(grep -q "ENABLE_METRICS_SERVER=true" ${CNODE_HOME}/mithril/mithril.env && echo "true" || echo "false")
    if [[ "${signerMetricsEnabled}" == "true" ]] ; then
      mithrilSignerMetrics=$(curl -s "http://${METRICS_SERVER_IP}:${METRICS_SERVER_PORT}/metrics" 2>/dev/null | grep -v -E "HELP|TYPE" | sed 's/mithril_signer_//g')
      SIGNER_METRICS_HTTP_RESPONSE=$(curl --write-out "%{http_code}" --silent --output /dev/null --connect-timeout 2 http://${METRICS_SERVER_IP}:${METRICS_SERVER_PORT}/metrics)
      if [[ "$SIGNER_METRICS_HTTP_RESPONSE" -eq 200 ]] ; then
        signerServiceStatus='online'
      else
        signerServiceStatus='offline'
      fi
      unset SIGNER_METRICS_HTTP_RESPONSE
    fi
  fi
}

```